### PR TITLE
Load DATABASE settings from YAML config (closes #12)

### LINKED ISSUE #12: Create configuration file
# Summary

There are a number of variables spread across this project that would be best abstracted out into a single YAML file at the root of the project, such as db password and username, allowed hosts, project name, etc

# Details

ideally, we would use a YAML file and add code into the relevant places to read in the variables using the python YAML library

# Outcome

All of our system-wide variables such as passwords or hosts or search tags should be in a single location that is easy to edit.

### PR DESCRIPTION
**Pull request by justwheel**
This pull request makes an assortment of changes. Git commit messages repeated below. When merging this PR, please use the "rebase" merge option. Thanks!

---

**Remove ambiguous config directory** — 5db1a0e

This commit removes the `config/` directory and the various web server
configs it included. Since we aren't using them and have not made an
intent on using them, it is confusing to leave them here while adding
other config files into the project.

---

**dependencies: Add pyyaml as a dependency for parsing YAML** — d567939

This commit was created by `pipenv install pyyaml`. This installs pyyaml
into the Pipfile so it is installed and built alongside our container.

---

**config: Add example config file and gitignore rule** — 87fe7bf

---

**Dockerfile: Remove mysqlclient hack step** — 752ded0

This commit reduces a hacky workaround we added for the mysqlclient
library. At the time, Pipenv could not install it but this is no longer
the case.

--- 

**settings: Load DATABASE settings from YAML config (closes #12)** — 3431e85

This commit removes the `DATABASE` options from the `development.py` and
`production.py` files and moves it into the `base.py` file. Instead of
hard-coding them into the application, the config values are retrieved
from a YAML config file at the root of the project directory.

Because the `config.yml` path is ignored, downstream users can copy the
`config.yml.example` file to `config.yml` and edit it freely as needed,
no matter what environment you are using or where your database is.

Closes #12.

---

**settings: Load YAML config path by CONFIGPATH env variable** — 69e11e9

This commit adds an option to specify a path to a config file as an
environment variable. If an environment variable is not set, the app
falls back on the `config.yml` at the root of the project directory.

This allows us some flexibility in how we end up deploying this in the
future and how we write our documentation.

### CODE REVIEW

**Comment by R-Eli-Levasseur** [comment]
Looking through the files changed it might be best to get Nathaniel's input on this one, nothing looks wrong to me though

File: .gitignore [hunk 1]
```diff
@@ -217,8 +228,7 @@ venv.bak/
 .dmypy.json
 dmypy.json
 
-#Database
-/db
```
Comment (nlMeminger) [inline .gitignore:221]: Why is the database path no longer ignored?

**Review by leong96 — APPROVED** [review]
Okay, that makes sense for how to read a yaml file from the settings.

File: .gitignore (hunk 1 above)
Comment (justwheel) [reply .gitignore:221]: @nlMeminger It is still ignored:

https://github.com/jwflory/django-rit-grasa/pull/66/files/69e11e94e3659973fc32c2c9a1b62f8423358461#diff-a084b794bc0759e7a6b77810e01874f2R5-R6

I regrouped our project-related paths together.

**Pull request merged**